staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -290,6 +291,11 @@ func patchResource(
 			// 3. ensure no conflicts between the two patches
 			// 4. apply the #1 patch to the currentJS object
 
+			// Don't calculate strategic merge patch if it is not supported. Instead, return early indicating a conflict.
+			if !sets.NewString(supportedPatchTypes...).Has(string(types.StrategicMergePatchType)) {
```

Comment:
we don't actually know if there's a meaningful conflict here... just that the resource version conflicted.

Reply:
Updated.